plugins/inputs/jolokia2_agent/jolokia2_agent_test.go

```diff
@@ -765,15 +765,6 @@ func TestFillFields(t *testing.T) {
 	require.Equal(t, map[string]interface{}{}, results)
 }
 
-func setupServer(resp string) *httptest.Server {
```

Comment:
The function wraps another function without adding _any_ added value, so why not moving the original code directly to the test so that you don't have to scroll trough the code to see what is going on?

Reply:
It's not just wrapping a function call, there's an anonymous func declaration, etc.

For me, the value is not needing to repeat those four lines seven times. That makes it clear at a glance that every test sets up its server in the same way. Without it, you have to visually compare the 4x7 lines to see if they are the same or if there's a subtle difference somewhere. Also, if there is ever a need to change server setup, it makes it so you don't have to do it in seven places.

I'm a fan of the [rule of three](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)) and this is complex enough and repeated enough that the rule applies.

Reply:
Ok will revert then.